### DISCUSSION TITLE
Does poetry not find the maximum supported version for a package it depends on?

### DISCUSSION THREAD

**Discussion by danielniccoli** [python-poetry/poetry#7023] [General] [answered]
I have a project that still runs on Django 2.2 and Python 3.7. (Will be changed soon, just not now.) I'm trying to integrate poetry.

I called `poetry init` and specified the python version as 3.7 (maximum supported version [as per Django docs](https://docs.djangoproject.com/en/4.0/faq/install/)).

```
[tool.poetry.dependencies]
python = "~3.7"
django = "~2.2"

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

Next, I call `poetry add django-registration`.

## Actual behaviour

It says that "no versions of django-registration match >3.3,<4.0" and that django-registration 3.3 depends on Django >= 3.2.

```
Using version ^3.3 for django-registration

Updating dependencies
Resolving dependencies...

Because no versions of django-registration match >3.3,<4.0
 and django-registration (3.3) depends on Django (>=3.2), django-registration (>=3.3,<4.0) requires Django (>=3.2).
So, because djangotest depends on both django (~2.2) and django-registration (^3.3), version solving failed.
```

## Expected behaviour

I didn't specify 3.3 as the minimum version for django-registration. I was expecting it to install version 3.2 as this is the last package that still has Django 2.2 listed as a supported package on the [PyPI page for the package](https://pypi.org/project/django-registration/3.2/).

When it checked that the most recent version was incompatible, I would expect poetry to check the next lower version. Does poetry not do that? Or am I missing something?

**Comment by radoering** [comment, marked as answer]
cf #707 and its duplicates
